contracts/StableJoeStaking.sol

```diff
@@ -0,0 +1,312 @@
+
+        uint256 rewardBalance;
+        if (_token == joe) rewardBalance = _token.balanceOf(address(this)).sub(internalJoeBalance);
+        else rewardBalance = _token.balanceOf(address(this));
+        if (rewardBalance != lastRewardBalance[_token] && totalJoe != 0) {
+            uint256 accruedReward = rewardBalance.sub(lastRewardBalance[_token]);
+            _accRewardTokenPerShare = _accRewardTokenPerShare.add(accruedReward.mul(PRECISION).div(totalJoe));
+        }
+        return user.amount.mul(_accRewardTokenPerShare).div(PRECISION).sub(user.rewardDebt[_token]);
+    }
+
+    /**
+     * @notice Withdraw JOE from sJOE and harvest the rewards
+     * @param _amount The amount of JOE to withdraw
+     */
+    function withdraw(uint256 _amount) external {
+        UserInfo storage user = userInfo[msg.sender];
+        uint256 previousAmount = user.amount;
+        require(previousAmount >= _amount, "StableJoeStaking: withdraw amount exceeds balance");
+        uint256 newAmount = user.amount.sub(_amount);
+        user.amount = newAmount;
+
+        uint256 len = rewardTokens.length;
+        uint256 pending;
+        if (previousAmount > 0) {
+            for (uint256 i; i < len; i++) {
+                IERC20Upgradeable token = rewardTokens[i];
+                updateReward(token);
+
+                pending = previousAmount.mul(accRewardPerShare[token]).div(PRECISION).sub(user.rewardDebt[token]);
+                if (pending != 0) safeTokenTransfer(token, msg.sender, pending);
+                user.rewardDebt[token] = newAmount.mul(accRewardPerShare[token]).div(PRECISION);
+            }
+        }
+
+        internalJoeBalance = internalJoeBalance.sub(_amount);
+        joe.safeTransfer(msg.sender, _amount);
+        emit Withdraw(msg.sender, _amount);
+    }
+
+    /**
+     * @notice Withdraw without caring about rewards. EMERGENCY ONLY
+     */
+    function emergencyWithdraw() external {
+        UserInfo storage user = userInfo[msg.sender];
+
+        uint256 _amount = user.amount;
+        user.amount = 0;
+        uint256 len = rewardTokens.length;
+        for (uint256 i; i < len; i++) {
+            IERC20Upgradeable token = rewardTokens[i];
+            user.rewardDebt[token] = 0;
+        }
+        joe.safeTransfer(msg.sender, _amount);
+        emit EmergencyWithdraw(msg.sender, _amount);
+    }
+
+    /**
+     * @notice Update reward variables to be up-to-date
+     * @param _token The address of the reward token
+     * @dev Needs to be called before any deposit or withdrawal
+     */
+    function updateReward(IERC20Upgradeable _token) public {
+        require(isRewardToken[_token], "StableJoeStaking: wrong reward token");
+        uint256 rewardBalance;
+        if (_token == joe) rewardBalance = _token.balanceOf(address(this)).sub(internalJoeBalance);
+        else rewardBalance = _token.balanceOf(address(this));
+        uint256 totalJoe = internalJoeBalance;
```

Comment:
I think you can get rid of this `totalJoe` variable and just use `internalJoeBalance` on lines 278 and 284.

Reply:
Same reason as before. Tho I don't know if it's worth here as we only use it twice.